interfaces/builtin/posix_mq.go

```diff
@@ -0,0 +1,237 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/seccomp"
+	"github.com/snapcore/snapd/snap"
+)
+
+const posixMQSummary = `allows access to POSIX message queues`
+
+const posixMQBaseDeclarationSlots = `
```

Comment:
@alexmurray if I understand correctly this has a flat namespace like shared-memory so it needs to be superprivileged so that we can manage the shared namespace

Reply:
ACK - will remove

Reply:
@pedronis should both the plug and slot be set to `allow-installation: false` or just the slot, like in shared-memory?

Reply:
I think the reasoning for shared-memory is the same here

Reply:
@pedronis thanks for the hint re managing the namespace - I hadn't thought of that - I wonder if we should document this somewhere (ie. interfaces which provide device ownership *or* which have a shared namespace should be super-privileged) or somesuch?